dual_net.py

```diff
@@ -305,30 +293,31 @@ def export_model(working_dir, model_path):
         tf.gfile.Copy(filename, destination_path)
 
 
-def train(working_dir, tf_records, steps=None, **hparams):
-    estimator = get_estimator(working_dir, **hparams)
+def train(working_dir, tf_records, steps=None):
+    estimator = get_estimator(working_dir)
 
     def input_fn():
-        return preprocessing.get_input_tensors(TRAIN_BATCH_SIZE, tf_records,
-                                               filter_amount=1.0)
+        return preprocessing.get_input_tensors(
+            FLAGS.train_batch_size, tf_records, filter_amount=1.0,
+            shuffle_buffer_size=FLAGS.shuffle_buffer_size)
 
     update_ratio_hook = UpdateRatioSessionHook(working_dir)
     step_counter_hook = EchoStepCounterHook(output_dir=working_dir)
     if steps is None:
-        steps = EXAMPLES_PER_GENERATION // TRAIN_BATCH_SIZE
+        steps = EXAMPLES_PER_GENERATION // FLAGS.train_batch_size
     estimator.train(input_fn, hooks=[
                         update_ratio_hook, step_counter_hook], steps=steps)
 
 
-def validate(working_dir, tf_records, checkpoint_name=None, **hparams):
-    estimator = get_estimator(working_dir, **hparams)
+def validate(working_dir, tf_records, checkpoint_name=None, validate_name=None):
+    estimator = get_estimator(working_dir)
     if checkpoint_name is None:
         checkpoint_name = estimator.latest_checkpoint()
 
     def input_fn():
-        return preprocessing.get_input_tensors(VALIDATE_BATCH_SIZE, tf_records,
+        return preprocessing.get_input_tensors(FLAGS.train_batch_size, tf_records,
```

Comment:
This is an intentional change but I think it should not affect things. Validation will now execute at the same batch size as training; since validation doesn't cause variable updates it should be irrelevant. It might actually be slower because it seems like we're doing steps=500, so we'd do 8x more validation.

Reply:
i'm fine with it, but validation takes a couple minutes right now.  i don't want it to be > 5min.